src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Collections/IIntervalTree`1.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.Shared.Collections;
+
+/// <summary>
+/// Generic function representing the type of interval testing operation that can be performed on an interval tree. For
+/// example checking if an interval 'contains', 'intersects', or 'overlaps' with a requested span.
+/// </summary>
+internal delegate bool TestInterval<T, TIntrospector>(T value, int start, int length, in TIntrospector introspector)
+    where TIntrospector : struct, IIntervalIntrospector<T>;
```

Comment:
new core types taht power the BinaryIntervalTree, and will power the upcoming FlatArrayIntervalTree.